application/src/main/java/org/togetherjava/tjbot/features/mediaonly/MediaOnlyChannelListener.java

```diff
@@ -54,6 +77,25 @@ private boolean messageHasNoMediaAttached(Message message) {
                 && !message.getContentRaw().contains("http");
     }
 
+
+    private RestAction<Message> warnUser(Message message) {
+        String originalMessageContent = message.getContentRaw();
+        MessageEmbed originalMessageEmbed =
+                new EmbedBuilder().setDescription(originalMessageContent)
+                    .setColor(Color.ORANGE)
+                    .build();
+
+        long authorId = message.getAuthor().getIdLong();
+
+        MessageCreateData pingMessage = new MessageCreateBuilder().setContent("Hey there, you <@"
+                + authorId
+                + "> posted a message without media (image, video, link) in a media-only channel. Please see the description of the channel for details and then repost with media attached, thanks 😀")
+            .setEmbeds(originalMessageEmbed)
+            .build();
+
+        return message.getChannel().sendMessage(pingMessage);
+    }
+
```

Comment:
instead modify the dmUser method to not dm but just give u the message to send, then once send it in dm and in the other place send it in the channel. So u won't have lots of copy pasted code

Reply:
No It requires a ping and Different Kind of string so I can make a common method to return Embed if its Efficient as its the only common code

Reply:
> No It requires a ping

just add a ping at the end or at the start on the common message

> Different Kind of string

??

Reply:
different string (not kind of).
UX can be updated later as required also I am not sure of the contents to bewritten there